aesara/tensor/subtensor_opt.py

```diff
@@ -1646,23 +1646,18 @@ def local_subtensor_SpecifyShape_lift(fgraph, node):
         return False
 
     obj_arg = specify_shape_node.owner.inputs[0]
-    shape_arg = specify_shape_node.owner.inputs[1]
+    shape_arg = specify_shape_node.owner.inputs[1:]
 
     indices = get_idx_list(node.inputs, node.op.idx_list)
 
-    if len(indices) > 1 or len(indices) == 0:
-        return False
-
-    if isinstance(indices[0], slice) or isinstance(
-        getattr(indices[0], "type", None), SliceType
+    if any(
```

Comment:
We could also handle mixed `[:]` slices easily, where we just have to keep the respective dimensions shape, but that's a bit beyond the  scope of this PR. 

Something like 
```python
# before:
specify_shape(x, (s1, s2))[:, 0 ]

#after:
specify_shape(x[:, 0], (s1,))
```